cmake/CMakeLists.txt

```diff
@@ -193,7 +193,8 @@ if(onnxruntime_USE_VALGRIND AND NOT WIN32)
 endif()
 
 if (onnxruntime_ENABLE_NVTX_PROFILE)
-  add_definitions(-DENABLE_NVTX_PROFILE=1)
+# TODO: This doesn't work with the shared cuda provider. Disabling temporarily to do a clean fix later as it wasn't trivial
+#  add_definitions(-DENABLE_NVTX_PROFILE=1)
```

Comment:
should we warn or error out early (temporarily) instead if onnxruntime_ENABLE_NVTX_PROFILE is set?
otherwise it just silently doesn't work as advertised

Reply:
Warning would be good, if it errors it'll just break the pipeline that has it set.

Reply:
Warning in place